scripts/experiments_tabular/knn.jl

```diff
@@ -80,33 +80,36 @@ end
 
 ####################################################################
 ################ THIS PART IS COMMON FOR ALL MODELS ################
-# paths
-savepath = datadir("experiments/tabular/$(modelname)/$(dataset)/seed=$(seed)") 
-mkpath(savepath)
-
-# get params, initialize the model, train it, predict scores and save everything
-data = GenerativeAD.load_data(dataset, seed=seed)
-
 # set a maximum for parameter sampling retries
 try_counter = 0
-max_tries = 10
-while try_counter < max_tries 
-	parameters = sample_params()
-	# here, check if a model with the same parameters was already tested
-	if GenerativeAD.check_params(GenerativeAD.edit_params, savepath, data, parameters)
-		# fit
-		training_info, results = fit(data, parameters)
-		# here define what additional info should be saved together with parameters, scores, labels and predict times
-		save_entries = merge(training_info, (modelname = modelname, seed = seed, dataset = dataset))
-		
-		# now loop over all anomaly score funs
-		for result in results
-			GenerativeAD.experiment(result..., data, savepath; save_entries...)
+max_tries = 10*max_seed
+while try_counter < max_tries
+    parameters = sample_params()
+
+    for seed in 1:max_seed
+		savepath = datadir("experiments/tabular/$(modelname)/$(dataset)/seed=$(seed)")
+		mkpath(savepath)
+
+		# get data
+		data = GenerativeAD.load_data(dataset, seed=seed)
+
+		# check if a combination of parameters and seed alread exists
+		if GenerativeAD.check_params(GenerativeAD.edit_params, savepath, data, parameters)
+			# fit
+			training_info, results = fit(data, parameters)
+			# here define what additional info should be saved together with parameters, scores, labels and predict times
+			save_entries = merge(training_info, (modelname = modelname, seed = seed, dataset = dataset))
+
+			# now loop over all anomaly score funs
+			for result in results
+				GenerativeAD.experiment(result..., data, savepath; save_entries...)
+			end
+			global try_counter = max_tries + 1
```

Comment:
`try_counter = max_tries` should be enough to terminate the while cycle

Reply:
Yes, but that triggers the printout on the last line, which should not happen if everything went alright

Reply:
Oh I see.